indexer/services/roundtable/src/tasks/aggregate-trading-rewards.ts

```diff
@@ -1,63 +1,230 @@
+import {
+  ONE_DAY_IN_MILLISECONDS,
+  ONE_MINUTE_IN_MILLISECONDS,
+  floorDate,
+  logger,
+} from '@dydxprotocol-indexer/base';
 import {
   BlockFromDatabase,
   BlockTable,
+  IsoString,
+  TradingRewardAggregationFromDatabase,
+  TradingRewardAggregationPeriod,
+  TradingRewardAggregationTable,
   TradingRewardFromDatabase,
 } from '@dydxprotocol-indexer/postgres';
-import { DateTime } from 'luxon';
+import { AggregateTradingRewardsProcessedCache } from '@dydxprotocol-indexer/redis';
+import { DateTime, Interval } from 'luxon';
+
+import { redisClient } from '../helpers/redis';
+import { UTC_OPTIONS } from '../lib/constants';
 
 /**
  * Task: Aggregate Trading Rewards
  * Description: This task aggregates trading rewards for a specific period of time.
  * It retrieves trading data from the database, calculates the rewards, and stores the aggregated
  * results.
  */
-interface Interval {
-  start: DateTime;
-  end: DateTime;
-}
-
 interface SortedTradingRewardData {
   [address: string]: TradingRewardFromDatabase[];
 }
 
-export default async function runTask(): Promise<void> {
-  // TODO(IND-499): Add resetting aggregation data when cache is empty
-  const interval: Interval | undefined = await getTradingRewardDataToProcessInterval();
-
-  const tradingRewardData: TradingRewardFromDatabase[] = await getTradingRewardDataToProcess(
-    interval,
-  );
-  const sortedTradingRewardData: SortedTradingRewardData = sortTradingRewardData(tradingRewardData);
-  await updateTradingRewardsAggregation(sortedTradingRewardData);
-  // TODO(IND-499): Update AggregateTradingRewardsProcessedCache
+export default function generateTaskFromPeriod(
+  period: TradingRewardAggregationPeriod,
+): () => Promise<void> {
+  return async () => {
+    const aggregateTradingReward: AggregateTradingReward = new AggregateTradingReward(period);
+    await aggregateTradingReward.runTask();
+  };
 }
 
-async function getTradingRewardDataToProcessInterval(): Promise<Interval> {
-  const latestBlock: BlockFromDatabase = await BlockTable.getLatest();
+export class AggregateTradingReward {
+  period: TradingRewardAggregationPeriod;
 
-  // TODO(IND-499): Setup AggregateTradingRewardsProcessedCache for start time and add end time
-  return {
-    start: DateTime.fromISO(latestBlock.time),
-    end: DateTime.fromISO(latestBlock.time),
-  };
-}
+  constructor(period: TradingRewardAggregationPeriod) {
+    this.period = period;
+  }
 
-async function getTradingRewardDataToProcess(
-  _interval: Interval,
-): Promise<TradingRewardFromDatabase[]> {
-  // TODO: Implement
-  return Promise.resolve([]);
-}
+  async runTask(): Promise<void> {
+    const interval: Interval | undefined = await this.getTradingRewardDataToProcessInterval();
+    if (interval === undefined) {
```

Comment:
if you get rid of the `getLatestBlock` exception handling in `getTradingRewardDataToProcessInterval`, you can simplify this code here to just `Interval` type